flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsStateBackend.java

```diff
@@ -141,6 +141,12 @@ public class FsStateBackend extends AbstractFileStateBackend implements Configur
      */
     private final int writeBufferSize;
 
+    /**
+     * Switch to create checkpoint sub-directory with name of jobId. A value of 'undefined' means
+     * not yet configured, in which case the default will be used.
+     */
+    private TernaryBoolean createCheckpointSubDirs = TernaryBoolean.UNDEFINED;
```

Comment:
I wonder do we really need to introduce an undefined Boolean here in such way.
Currently, we only have one way to set this value within a private constructor:
~~~java
this.createCheckpointSubDirs =
    original.createCheckpointSubDirs.resolveUndefined(
        configuration.get(CheckpointingOptions.CREATE_CHECKPOINT_SUB_DIS));
~~~

However, there is no public constructor or setter to set the `createCheckpointSubDirs` in the `original` one, that is to say we will always get a `TernaryBoolean.UNDEFINED` `createCheckpointSubDirs` in the original.

Reply:
Good point, I removed the TernaryBoolean here.